app/jobs/create_derivatives_job_decorator.rb

```diff
@@ -35,6 +31,21 @@ def perform(file_set, file_id, filepath = nil)
       file_set.reload
       file_set.update_index
       file_set.parent.update_index if parent_needs_reindex?(file_set)
+    rescue StandardError => e
+      if time_to_live.positive?
+        # It's possible we can recover from this, so we'll give it another go.
+        logger.warning(
+          "WARNING: FileSet ID=\"#{file_set_id}\" error for #{self.class}: #{e}. " \
+          "Retries remaining #{time_to_live - 1}."
+        )
+        CreateDerivativesJob.perform_later(file_set, file_id, filepath, time_to_live - 1)
+      else
+        logger.error(
+          "ERROR: FileSet ID=\"#{file_set_id}\" error for #{self.class}: #{e}. " \
+          "No retries remaining.  Backtrace: #{e.backtrace}"
+        )
+      end
+      return false
```

Comment:
Do we want this to `return false` in all cases?  My assumption would be that we would return false if we reschedule or encounter an error.  But return true if we complete successfully.

All of which might be mute because the return value may be inconsequential.

Reply:
I think I misunderstood your original suggestion and took it too literal. I will make changes again and use the referenced source as inspiration :)